plugins/woocommerce/tests/e2e-pw/tests/merchant/create-simple-product.spec.js

```diff
@@ -107,7 +96,11 @@ test.describe( 'Add New Simple Product Page', () => {
 		await expect(
 			page.locator( 'a.shipping-calculator-button' )
 		).not.toBeVisible();
-		await page.click( 'a.remove' );
+		await page.click( `a.remove[data-product_id='${ virtualProductId }']` );
+		await page.waitForLoadState( 'networkidle' );
+		await expect(
+			page.locator( `a.remove[data-product_id='${ virtualProductId }']` )
+		).not.toBeVisible();
```

Comment:
Sometimes, the test clicks the "remove" button of a different product. This change makes sure that it removes the correct product using its ID, and then asserts that it's successfully removed.